api/nodes/serializers.py

```diff
@@ -935,3 +937,120 @@ def update(self, draft, validated_data):
             draft.update_metadata(metadata)
             draft.save()
         return draft
+
+
+class NodeVOL(ser.Field):
+    def to_representation(self, obj):
+        if obj is not None:
+            return obj._id
+        return None
+
+    def to_internal_value(self, data):
+        return data
+
+
+class NodeViewOnlyLinkSerializer(JSONAPISerializer):
+    key = ser.CharField(read_only=True)
+    id = IDField(source='_id', read_only=True)
+    date_created = ser.DateTimeField(read_only=True)
+    anonymous = ser.BooleanField(required=False)
+    name = ser.CharField(required=False)
+
+    creator = RelationshipField(
+        related_view='users:user-detail',
+        related_view_kwargs={'user_id': '<creator._id>'},
+    )
+
+    nodes = RelationshipField(
+        related_view='view_only_links:view-only-link-nodes',
+        related_view_kwargs={'link_id': '<_id>'}
+    )
+
+    def get_absolute_url(self, obj):
+        return absolute_reverse(
+            'nodes:node-view-only-link-detail',
+            kwargs={
+                'key': obj.key
+            }
+        )
+
+    class Meta:
+        type_ = 'view_only_links'
+
+
+class NodeViewOnlyLinkUpdateSerializer(NodeViewOnlyLinkSerializer):
+    """
+    Overrides NodeViewOnlyLinkSerializer to make nodes a list field.
+    """
+    nodes = JSONAPIListField(child=NodeVOL(), required=False)
+
+    def update(self, link, validated_data):
+        assert isinstance(link, PrivateLink), 'link must be a PrivateLink'
+        return self.update_view_only_link(
+            view_only_link=link,
+            user=get_user_auth(self.context['request']).user,
+            name=validated_data.get('name'),
+            anonymous=validated_data.get('anonymous'),
+            nodes=validated_data.get('nodes')
+        )
+
+    def update_view_only_link(self, view_only_link, user, name=None, anonymous=None, nodes=None):
+        if name:
+            name = strip_html(name)
+            if name is None or not name.strip():
+                raise exceptions.ValidationError('Invalid link name.')
+            view_only_link.name = name
+
+        if anonymous:
+            view_only_link.anonymous = anonymous
+
+        if nodes:
```

Comment:
I suspect this whole serializer is not necessary. Ideally the relationship self link (that you're learning about in your other issue) will handle managing the nodes that are related to this VOL, and ideally ideally the relationshipfield would be able to handle being given a list of items. That being said, I'm not entirely sure that all of the plumbing is in for this. Hmm.

While QA is working on the tests for this, we might want to take a look at what the Preprints team did to update the relationship link management and see if it's partially, fully, or not at all useful for this. Then we can decide if we want to make a TODO to make this work better or if we want to just go ahead and make it right now.